src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstancesController.cs

```diff
@@ -146,10 +146,18 @@ public class InstancesController : ControllerBase
 
                 if (string.IsNullOrEmpty(org) && string.IsNullOrEmpty(appId))
                 {
-                    return BadRequest("Org or AppId must be defined.");
+                    return BadRequest("org or appId must be defined.");
                 }
 
-                org = string.IsNullOrEmpty(org) ? appId.Split('/')[0] : org;
+                if (!string.IsNullOrEmpty(appId))
+                {
+                    var appParts = appId.Split('/', 2);
+                    if(appParts.Length != 2 || (string.IsNullOrEmpty(org) && appParts[0] != org))
```

Comment:
Should this have been _not null or empty_?

Reply:
I should have a pre commit hook that issues a warning when i reach for `isset` and forget to write the `!` in front of `string.isNullOrEmpty`.
```suggestion
                    if(appParts.Length != 2 || (!string.IsNullOrEmpty(org) && appParts[0] != org))
```